spec/models/theme_spec.rb

```diff
@@ -68,14 +68,50 @@
   end
 
   it 'can correctly find parent themes' do
-    grandchild = Theme.create!(name: 'grandchild', user_id: user.id)
-    child = Theme.create!(name: 'child', user_id: user.id)
-    theme = Theme.create!(name: 'theme', user_id: user.id)
+    child = Theme.create!(name: 'child', user: user)
+    theme = Theme.create!(name: 'theme', user: user)
 
     theme.add_child_theme!(child)
-    child.add_child_theme!(grandchild)
 
-    expect(grandchild.dependant_themes.length).to eq(2)
+    expect(child.dependant_themes.length).to eq(1)
+  end
+
+  it "doesn't allow multi-level theme components" do
+    grandchild = Theme.create!(name: 'grandchild', user: user)
+    child = Theme.create!(name: 'child', user: user)
+    theme = Theme.create!(name: 'theme', user: user)
+    grandparent = Theme.create!(name: 'grandparent', user: user)
+
+    theme.add_child_theme!(child)
+    expect do
+      child.add_child_theme!(grandchild)
+    end.to raise_error(Discourse::InvalidParameters, I18n.t("themes.errors.no_multilevels_components"))
+
+    expect do
+      grandparent.add_child_theme!(theme)
+    end.to raise_error(Discourse::InvalidParameters, I18n.t("themes.errors.no_multilevels_components"))
+  end
+
+  it "doesn't allow a child to be user selectable" do
+    child = Theme.create!(name: 'child', user: user)
+    theme = Theme.create!(name: 'theme', user: user)
+
+    theme.add_child_theme!(child)
+    begin
+      child.update!(user_selectable: true)
+    rescue ActiveRecord::RecordInvalid => e
```

Comment:
We don't need to `rescue` the error here as we can use `update` instead of `update!`